packages/taquito-local-forging/src/proto13-jakarta/michelson-proto13/codec-proto13.ts

```diff
@@ -0,0 +1,176 @@
+import { Decoder } from '../../decoder';
+import { pad } from '../../utils';
+import { Encoder } from '../../encoder';
+import {
+  bytesDecoder,
+  bytesEncoder,
+  decodeAnnots,
+  encodeAnnots,
+  extractRequiredLen,
+  intDecoder,
+  intEncoder,
+  isBytes,
+  isInt,
+  isPrim,
+  isString,
+  MichelsonValue,
+  PrimValue,
+  stringDecoder,
+  stringEncoder,
+} from '../../michelson/codec';
+import { Uint8ArrayConsumer } from '../../uint8array-consumer';
+import { opMappingProto13, opMappingReverseProto13 } from '../constants-proto13';
+
+export const scriptEncoderProto13: Encoder<{ code: MichelsonValue; storage: MichelsonValue }> = (
+  script
+) => {
+  const code = valueEncoderProto13(script.code);
+  const storage = valueEncoderProto13(script.storage);
+  return `${pad(code.length / 2, 8)}${code}${pad(storage.length / 2, 8)}${storage}`;
+};
+
+export const scriptDecoderProto13: Decoder = (value: Uint8ArrayConsumer) => {
+  const code = extractRequiredLen(value);
+  const storage = extractRequiredLen(value);
+
+  return {
+    code: valueDecoderProto13(new Uint8ArrayConsumer(code)),
+    storage: valueDecoderProto13(new Uint8ArrayConsumer(storage)),
+  };
+};
+
+export const valueEncoderProto13: Encoder<MichelsonValue> = (value: MichelsonValue) => {
+  if (Array.isArray(value)) {
+    const encoded = value.map((x) => valueEncoderProto13(x)).join('');
+    const len = encoded.length / 2;
+    return `02${pad(len)}${encoded}`;
+  } else if (isPrim(value)) {
+    return primEncoderProto13(value);
+  } else if (isBytes(value)) {
+    return bytesEncoder(value);
+  } else if (isString(value)) {
+    return stringEncoder(value);
+  } else if (isInt(value)) {
+    return intEncoder(value);
+  }
+
+  throw new Error('Unexpected value');
```

Comment:
I believe we have added a new error class in the error pattern PR called `UnexpectedMichelsonValueError` for this case